src/client.rs

```diff
@@ -434,20 +438,117 @@ impl Client {
     }
 }
 
+pub struct ClientBuilder<C: Connectable> {
+    target: C,
+    max_size: u32,
+    min_idle: Option<u32>,
+    max_lifetime: Option<Duration>,
+    read_timeout: Option<Duration>,
+    write_timeout: Option<Duration>,
+    pool_wait_timeout: Duration,
+}
+
+impl<C: Connectable> ClientBuilder<C> {
+    pub fn new(target: C) -> Self {
+        ClientBuilder {
+            target,
+            max_size: 1,
+            min_idle: None,
+            max_lifetime: None,
+            read_timeout: None,
+            write_timeout: None,
+            pool_wait_timeout: Duration::from_secs(30),
+        }
+    }
+
+    pub fn with_max_size(mut self, max_size: u32) -> Self {
+        self.max_size = max_size;
+        self
+    }
+
+    pub fn with_min_idle(mut self, min_idle: u32) -> Self {
```

Comment:
I think these configurations is for r2d2 pool, but with the name `max_size` or `min_idle` is really confusing. Maybe some user will thinks the `max_size` is for configure the max item size in memcached.

So it will be more clear to add a prefix, like `pool_`.

Reply:
Makes sense, I'll change that and take another look at the rest of the names